src/disambiguate-series-statements.js

```diff
@@ -0,0 +1,218 @@
+    const recordType = getRecordType(record);
+    //console.info(`getDeletableFields(record), type: ${recordType}`); // eslint-disable-line no-console
+    if (recordType === UNDEFINED) {
+      return [];
+    }
+
+    const seriesStatements = record.get('490').filter(f => fieldHasSubfield(f, 'x'));
+
+    //console.info(`N STATEMENTS=${seriesStatements.length}`); // eslint-disable-line no-console
+    if (seriesStatements.length < 2) {
+      return [];
+    }
+
+    const deletableFields = await getDeletableSeriesStatementFields(recordType, seriesStatements);
+    // Delete something if and only if we keep something as well:
+    if (deletableFields.length < seriesStatements.length) {
+      return deletableFields;
+    }
+
+    return [];
+  }
+
+  async function getDeletableSeriesStatementFields(recordType, seriesStatementFields, deletableFields = []) {
+    //console.info(`getDeletableSeriesStatementFields(), N CANDS=${seriesStatementFields.length}`); // eslint-disable-line no-console
+
+    if (seriesStatementFields.length === 0) {
+      //console.info(` DONE`); // eslint-disable-line no-console
+      return deletableFields;
+    }
+
+    const [currStatementField, ...rest] = seriesStatementFields;
+
+    const removeMe = await isRemovableField(currStatementField);
+    //console.info(` ${removeMe ? 'REMOVE' : 'KEEP'}: ${fieldToString(currStatementField)}`); // eslint-disable-line no-console
+
+    const deletableFields2 = removeMe ? [...deletableFields, currStatementField] : deletableFields;
+
+    const result = await getDeletableSeriesStatementFields(recordType, rest, deletableFields2);
+    return result;
+
+    async function isRemovableField(seriesStatementField) {
+      //console.info(` isRemovableField() in...`); // eslint-disable-line no-console
+      const issn = getIssn(seriesStatementField);
+      if (!issn) {
+        return false;
+      }
+      //console.info(` got ISSN ${issn}`); // eslint-disable-line no-console
+      const issnRecords = await issnToRecords(issn);
+      //console.info(` ISSN returned ${issnRecords.length} record(s)`); // eslint-disable-line no-console
+      // TEE: konvertoi tietueet objekteiksi!
+      if (issnRecords.some(r => !isMismatchingRecord(r))) {
+        return false;
+      }
+
+      return true;
+    }
+
+    function isMismatchingRecord(r) {
+      const issnRecordType = getRecordType(r);
+      if (issnRecordType === UNDEFINED) {
+        return false;
+      }
+      return issnRecordType !== recordType;
+    }
+  }
+
+  async function issnToRecords(issn) {
+    //console.log('issnToRecords() in...'); // eslint-disable-line no-console
+    const records = await search(sruClient, `bath.issn=${issn}`);
+    //console.log(`ISSN2RECORDS got ${records.length} record(s)!`); // eslint-disable-line no-console
+    return records;
+  }
+
+  function getIssn(field) {
+    // Assume that this is a field 490.
+    // Return value in first $x subfield:
```

Comment:
This logic needs to be expanded to managing multiple $x in one field

Reply:
Yes. My bad. Fixed in latest.